src/benchmarks/micro/runtime/BenchmarksGame/binarytrees-6.cs

```diff
@@ -0,0 +1,123 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+// Adapted from binary-trees C# .NET Core #6 program
+// https://salsa.debian.org/benchmarksgame-team/benchmarksgame/-/blob/master/public/download/benchmarksgame-sourcecode.zip
+// Best-scoring C# .NET Core version as of 2020-08-12
+
+// The Computer Language Benchmarks Game
+// https://salsa.debian.org/benchmarksgame-team/benchmarksgame/
+//
+// contributed by Marek Safar
+// concurrency added by Peperud
+// fixed long-lived tree by Anthony Lloyd
+// ported from F# version by Anthony Lloyd
+
+using System;
+using System.Runtime.CompilerServices;
+using System.Threading.Tasks;
+using BenchmarkDotNet.Attributes;
+using MicroBenchmarks;
+
+namespace BenchmarksGame
+{
+    [MaxIterationCount(40)] // the default 20 is not enough, the benchmark has multimodal distribution and needs more runs
+    [BenchmarkCategory(Categories.Runtime, Categories.BenchmarksGame)]
+    public class BinaryTrees_6
+    {
+        const int MinDepth = 4;
+        const int NoTasks = 4;
+
+        // 16 is about 80ms
+        // 18 is about 500ms
+        // 20 is about 3.9s
+        // 21 is used in official numbers; about 7.8s
+        const int N = 18;
```

Comment:
the two existign `BinaryTrees` benchmarks use `n=16`:

https://github.com/dotnet/performance/blob/d6ff9585a295f0b2d7570b54d84c24607480d661/src/benchmarks/micro/runtime/BenchmarksGame/binarytrees-2.cs#L28-L29

https://github.com/dotnet/performance/blob/d6ff9585a295f0b2d7570b54d84c24607480d661/src/benchmarks/micro/runtime/BenchmarksGame/binarytrees-5.cs#L33-L34

perhaps it would be better to use `16` here as well to allow for apples-to-apples comparison?

80 ms should not be a problem as BDN will try to figure out how many times it should be invoked to make sure that a single iteration lasts 250 ms
```suggestion
        // 16 is about 80ms
        // 18 is about 500ms
        // 20 is about 3.9s
        // 21 is used in official numbers; about 7.8s
        const int N = 16;
```

Reply:
For this and the suggestion of 10 below -- I am intentionally trying to use closer to what the benchmark does, since an eg 80ms operation (even repeated) may not have the same dynamics as the benchmark, which is a 7.8 sec operation, and 500ms may be closer. Eg., the cost of starting threads can be amortized. What do you think?

Reply:
You are right, being closer to what the Computer Games Benchmark does is more important than comparing the benchmark variants (BinaryTrees 2 vs BinaryTrees 6)